exp/api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -240,6 +240,24 @@ func (m *AzureManagedControlPlane) ValidateUpdate(oldRaw runtime.Object, client
 		}
 	}
 
+	// NOTE: This only works because we force the user to set the CIDRBlock for both the
```

Comment:
@jackfrancis @CecileRobertMichon Please note this caveat with this logic. Personally I consider the lack of reconciliation of the Subnet cidr block a bug and think it should be done like the AzureCluster, but I think for this purpose we should cross that bridge when it happens. Thoughts?

Reply:
Thanks for your stamina here @mtougeron, will get to a thorough review of this soon. ❤️